tests/e2e-pw/tests/price-filter/price-filter.page.ts

```diff
@@ -0,0 +1,173 @@
+			postId: 'woocommerce/woocommerce//archive-product',
+			postType: 'wp_template',
+		} );
+
+		await this.editor.canvas.click( 'body' );
+
+		await this.editor.insertBlock( {
+			name: WOOCOMMERCE_FILTER_WRAPPER,
+			attributes: {
+				filterType: 'price-filter',
+				heading: 'Filter By Price',
+			},
+		} );
+		await this.page.waitForLoadState( 'networkidle' );
+		await this.editor.saveSiteEditorEntities();
+		await this.page.goto( `/shop`, { waitUntil: 'networkidle' } );
+	}
+
+	async addPriceFilterBlockToNewPostAndGoToFrontend() {
+		await this.admin.createNewPost();
+		await this.editor.insertBlock( { name: WOOCOMMERCE_ALL_PRODUCTS } );
+		await this.editor.insertBlock( {
+			name: WOOCOMMERCE_FILTER_WRAPPER,
+			attributes: {
+				filterType: 'price-filter',
+				heading: 'Filter By Price',
+			},
+		} );
+		await this.editor.publishPost();
+		await this.page.waitForLoadState( 'networkidle' );
+		const url = new URL( this.page.url() );
+		const postId = url.searchParams.get( 'post' );
+		await this.page.goto( `/?p=${ postId }`, { waitUntil: 'networkidle' } );
+	}
+
+	async setPriceFilterRange(
+		minPrice: number | null,
+		maxPrice: number | null
+	) {
+		const priceFilterBlock = await this.page.locator( blockData.mainClass );
+		const minPriceInput = await priceFilterBlock.getByRole( 'textbox', {
+			name: 'Filter products by minimum price',
+		} );
+		const maxPriceInput = await priceFilterBlock.getByRole( 'textbox', {
+			name: 'Filter products by maximum price',
+		} );
+
+		if ( minPrice !== null ) {
+			await minPriceInput.click();
+			await minPriceInput.fill( `${ minPrice }` );
+		}
+		if ( maxPrice !== null ) {
+			await maxPriceInput.click();
+			await maxPriceInput.fill( `${ maxPrice }` );
+		}
+
+		await this.page.click( 'body' );
+
+		await this.page.waitForURL( ( url ) =>
+			url
+				.toString()
+				.includes(
+					this.generatePriceFilterQueryString( minPrice, maxPrice )
+				)
+		);
+	}
+
+	async revertArchiveProductTemplate() {
+		await this.templateApiUtils.revertTemplate(
+			'woocommerce/woocommerce//archive-product'
+		);
+	}
+
+	/**
+	 * Locators
+	 */
+	async locateFirstImage() {
+		await this.page.waitForLoadState( 'networkidle' );
+		const allProductsBlock = await getBlockByName( {
+			page: this.page,
+			name: WOOCOMMERCE_ALL_PRODUCTS,
+		} );
+
+		return await allProductsBlock.locator( 'img' ).first();
+	}
+
+	async locateAllProductsBlock() {
+		return await getBlockByName( {
+			page: this.page,
+			name: WOOCOMMERCE_ALL_PRODUCTS,
+		} );
+	}
+
+	async locateAllProducts() {
+		const products = await this.page.locator(
+			'.products-block-post-template .product'
+		);
+		return products;
+	}
+
+	/**
+	 * Private functions to be used in this class
+	 */
+	private generatePriceFilterQueryString(
+		minPrice: number | null,
+		maxPrice: number | null
+	) {
+		let result = '';
+		if ( minPrice ) {
+			result += `min_price=${ minPrice }`;
+		}
+		if ( maxPrice ) {
+			result += `&max_price=${ maxPrice }`;
```

Comment:
> Based on the output image it seems all good, but the event was not fired, so do you think it could be a race condition? 🤔

I am not sure because all tests are running successfully locally for me. I will try to reproduce the issue by following these steps:
1. Destroying the environment: `npm run wp-env destroy`
2. Start the environment again using: `npm run env:start`
3. Once the environment is up and running, re-run the tests using: `npm run test:e2e-pw`

I will update you once I have more information.

Reply:
> Detail, but in case there's no min_price query would start with an ampersand (&max_price). I don't know if that may influence the test results, but I'd keep it the closest to the real env possible. Unless you're sure that's ok.

Agreed, I made the changes in [8c3efcb](https://github.com/woocommerce/woocommerce-blocks/pull/10031/commits/8c3efcb206845d5ed68b8010e7d0b50e287dfc6d). Now function uses `URLSearchParams` to generate the query string. This object automatically encodes special characters and manages the & delimiter.